From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Tue, 22 Jun 2021 19:32:14 +0200
Applied-Upstream: https://github.com/logrotate/logrotate/commit/ec05c1b54919e60c5c686bb866f67591d2977f25
Subject: Remove invalid configuration on error

After failing while parsing an invalid configuration file, like:

    /some/path

remove the erroneous configuration structure from the internal list.

Else one might see:

    reading config file config.tmp
    error: config.tmp:1 missing '{' after log files definition
    Reading state from file: state.tmp
    error: error opening state file state.tmp: No such file or directory
    Allocating hash table for state file, size 64 entries

    Handling 1 logs

    rotating pattern: (null) forced from command line (no old logs will be kept)
    empty log files are rotated, old logs are removed
    No logs found. Rotation not needed.
---
 config.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/config.c b/config.c
index bbdbc45..ca32cd6 100644
--- a/config.c
+++ b/config.c
@@ -2113,6 +2113,8 @@ next_state: ;
     close(fd);
     return logerror;
 error:
+    if (newlog != defConfig)
+        freeTailLogs(1);
     /* free is a NULL-safe operation */
     free(key);
     munmap(buf, length);
